app/models/has_corporate_information_pages.rb

```diff
@@ -4,7 +4,7 @@ module HasCorporateInformationPages
   included do
     has_many :corporate_information_pages, through: "edition_#{table_name}".to_sym, source: :edition, class_name: "CorporateInformationPage"
     before_destroy do |record|
-      record.corporate_information_pages.map { |cip| cip.document.destroy }
+      record.corporate_information_pages.map(&:document).uniq.each(&:destroy)
```

Comment:
Ideally this would be transacted or the collection deleted atomically via `delete_all` or friends, on an arel.